code/_onclick/hud/radial.dm

```diff
@@ -18,7 +18,7 @@ GLOBAL_LIST_EMPTY(radial_menus)
 
 /atom/movable/screen/radial/proc/handle_parent_del()
 	SIGNAL_HANDLER
-	set_parent(null)
+	qdel(src) // No reason for us to exist if our parent menu's gone and we don't have a new one
```

Comment:
Results in extremely intense hard dels

Reply:
Ah, didn't see this happening when I tested it. I'll try something else to fix the issue then.